packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -38,31 +70,79 @@ const ContextMenu: React.FC<{ onOpenChange?(open: boolean): void }> = (props) =>
   );
 
   return (
-    <MenuPrimitive.Root open={open} onOpenChange={handleOpenChange}>
+    <MenuPrimitive.Root open={open} onOpenChange={handleOpenChange} dir={dir}>
       <ContextMenuProvider open={open} onOpenChange={handleOpenChange}>
         {children}
       </ContextMenuProvider>
     </MenuPrimitive.Root>
   );
 };
 
+/* -----------------------------------------------------------------------------------------------*/
+
+type ContextMenuSubOwnProps = {
+  open?: boolean;
+  onOpenChange?(open: boolean): void;
+  defaultOpen?: boolean;
+  dir?: Direction;
+};
+
+const ContextMenuSub: React.FC<ContextMenuSubOwnProps> = (props) => {
+  const { open: openProp, defaultOpen, onOpenChange, ...menuSubProps } = props;
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpen,
+    onChange: onOpenChange,
+  });
+
+  return <MenuPrimitive.Sub {...menuSubProps} open={open} onOpenChange={setOpen} />;
+};
+
 ContextMenu.displayName = CONTEXT_MENU_NAME;
 
 /* -------------------------------------------------------------------------------------------------
  * ContextMenuTrigger
  * -----------------------------------------------------------------------------------------------*/
 
 const TRIGGER_NAME = 'ContextMenuTrigger';
-const TRIGGER_DEFAULT_TAG = 'span';
+const ROOT_TRIGGER_DEFAULT_TAG = 'span';
```

Comment:
It only applies to the root trigger so figured it was more appropriate.

Reply:
I'm not sure what we usually do here, sometimes I've left the name as it was and just used it in the other part, or I've renamed and moved it alongside the other part.

@benoitgrelard what should we do here? I'm thinking move it but I've never been sure myself on this one haha.

Reply:
I actually have an inconsistency here because I moved it in `DropdownMenu` but not here 😅 I'm happy to keep the previous naming if our standard is to use proximity to define the relationship 👍

Reply:
I'm not sure I follow what you mean by "move" it.

Reply:
Moved the `const` to sit next to the component using it.

Reply:
ohhh I see, yeah generally I like to have these just above their own part.